tools/misc/split_coco.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import argparse
+import os.path as osp
+
+import mmcv
+import numpy as np
+
+
+def split_coco(data_root, out_dir, fold, percent):
+    """Split COCO data for Semi-supervised object detection.
+
+    Args:
+        data_root: root of dataset
+        fold: the fold of dataset and set as random seed for data split
+        percent: percentage of labeled data
+    """
+
+    def save_anno(name, images, annotations):
+        print(f'Starting to split data {name}.json '
+              f'saved {len(images)} images and {len(annotations)}annotations')
+
+        sub_annos = {}
+        sub_annos['images'] = images
+        sub_annos['annotations'] = annotations
+        sub_annos['licenses'] = annos['licenses']
+        sub_annos['categories'] = annos['categories']
+        sub_annos['info'] = annos['info']
+
+        mmcv.mkdir_or_exist(out_dir)
+        mmcv.dump(sub_annos, f'{out_dir}/{name}.json')
+
+        print(f'Finishing to split data {name}.json '
+              f'saved {len(images)} images and {len(annotations)} annotations')
+
+    np.random.seed(fold)
+    ann_file = osp.join(data_root, 'annotations/instances_train2017.json')
+    annos = mmcv.load(ann_file)
+
+    image_list = annos['images']
+    labeled_tot = int(percent / 100. * len(image_list))
+    labeled_ind = set(
+        np.random.choice(range(len(image_list)), size=labeled_tot))
```

Comment:
It is possible to choose the same index here, whether it is as expected？

Reply:
Following the  consistent logic  with Google's STAC and SoftTeacher here.